mutation/mutation_partition_v2.cc

```diff
@@ -110,9 +110,13 @@ void mutation_partition_v2::ensure_last_dummy(const schema& s) {
     }
 }
 
-std::ostream& operator<<(std::ostream& out, const apply_resume& res) {
-    return out << "{" << int(res._stage) << ", " << res._pos << "}";
-}
+template <>
+struct fmt::formatter<apply_resume> : fmt::formatter<std::string> {
```

Comment:
Why `fmt::formatter<std::string>` ?

Reply:
sorry, it was a mistake. changed to `string_view`.

Reply:
@tgrabiec hi Tomasz, see https://github.com/scylladb/scylladb/pull/13584#discussion_r1173323892 . if we really don't need the format specifier support, we can define a no-op `parse()` method. is this want you are proposing?

Reply:
I didn't mean to propose anything, it just seemed arbitrary to extend `fmt::formatter<std::string_view>` so I wondered what really is the intent. By Liskov substitution principle, this says that formatter of apply_resume is also a formatter of string_view, which is an unnecessary extension of responsibility.

Maybe this would express the intent cleaner:

```c++

struct with_default_format_specifier {
    constexpr auto parse(fmt::format_parse_context& ctx) {
       return fmt::formatter<std::string_view>()::parse(ctx);
    }
}

...


template <>
struct fmt::formatter<apply_resume> : with_default_format_specifier {
};
```

Reply:
i am following the document of https://fmt.dev/latest/api.html#formatting-user-defined-types . we can reuse the existing formatter either by composition or inheritance. in your code snippet, i am not sure which way you prefer, as `parse()` is not a static member function of `fmt::formatter<string_view>`.

`fmt::formatter<std::string_view>` is not randomly picked, its specifier is more likely to apply to other regular objects. as for instance, the `fmt::formatter<float>` would allow user to specify the precision. and `fmt::formatter<int>` allows user to specify if the number should be formatted in hex or oct, etc.

if we want to adhere to Liskov Substitution Principle here, we'd have to go with private inheritance or composition. either way makes the code more complicate than public inheritance. so i don't it is worthy to do this. as the way how the formatter is used is more in a domain of generic programming not object oriented programming, in the sense that fmtlib just instantiate an object of  `fmt::formatter<class_to_be_printed>`. what the parent class(es) of  `fmt::formatter<class_to_be_printed>` is does not really matter when the compiler performs the resolution.

Reply:
> i am not sure which way you prefer, as parse() is not a static member function of fmt::formatter<string_view>

In my snippet, I instantiate the formatter temporarily (so no static call).

>  we'd have to go with private inheritance or composition. either way makes the code more complicated than public inheritance

I proposed to use public inheritance from `with_default_format_specifier`

Reply:
> > i am not sure which way you prefer, as parse() is not a static member function of fmt::formatter<string_view>
> 
> In my snippet, I instantiate the formatter temporarily (so no static call).

then the `parse()` call would fail to persist the parsed specifiers, and the succeeding `format()` call would not be affected by the previously specified specifiers.

> 
> > we'd have to go with private inheritance or composition. either way makes the code more complicated than public inheritance
> 
> I proposed to use public inheritance from `with_default_format_specifier`

Reply:
> > > i am not sure which way you prefer, as parse() is not a static member function of fmt::formatter<string_view>
> > 
> > 
> > In my snippet, I instantiate the formatter temporarily (so no static call).
> 
> then the `parse()` call would fail to persist the parsed specifiers, and the succeeding `format()` call would not be affected by the previously specified specifiers.

There are no specifiers to persist for simple formatters. We could also do it like that, right?

```
struct basic_formatter {
     constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
}
```